Update API Docs to add Documentation on Mutations (#369)
ballerina-platform/module-ballerina-graphql: ballerina/Module.md

```

### Service
The Ballerina GraphQL service represents the GraphQL schema. When a service is attached to a `graphql:Listener`, a GraphQL schema will be auto-generated. 

> The GraphQL services are exposed through a single endpoint. The url of the GraphQL service can be provided when defining the service. (Check the next example, where the url is `/graphql`).

Reviewer: ```suggestion
The GraphQL services are exposed through a single endpoint. The URL of the GraphQL service can be provided when defining the service. (Check the next example, in which the URL is `/graphql`).
```

#### Query Type
The resource functions inside the service represent the resolvers of the `Query` root type.

> When a resource function is defined inside a GraphQL service, the generated schema will have a `Query` root type, and the resource function will be a field of the `Query` object.

Reviewer: ```suggestion
When a resource function is defined inside a GraphQL service, the generated schema will have a `Query` root type and the resource function will be a field of the `Query` object.
```

}
```

#### Mutation Type
The remote functions inside the GraphQL service represent the resolvers of the `Mutation` root type.

> When a remote function is defined inside a GraphQL service, the schema will have a `Mutation` operation, and the remote function will be a field of the `Mutation` object.

Reviewer: ```suggestion
When a remote function is defined inside a GraphQL service, the schema will have a `Mutation` operation and the remote function will be a field of the `Mutation` object.
```

> For an example, consider the following service, that has a `Person` record, `person`. It has a `Query` field named `profile`, which returns the person record. It also has two remote functions, named `updateName` and `updateCity`, which are used as mutations.

Reviewer: ```suggestion
For an example, consider the following service that has a `Person` record named `person`. It has a `Query` field named `profile`, which returns the `person` record. It also has two remote functions named `updateName` and `updateCity`, which are used as mutations.
```

```ballerina
import ballerina/graphql;

public type Person record {|
    string name;
    int age;
    string city;
|};

service /graphql on new graphql:Listener(4000) {
    private Person profile;
    
    function init() {
        self.profile = { name: "Walter White", age: 50, city: "Albuquerque" };
    }
    
    resource function get profile() returns Person {
        return self.profile;
    }
    
    remote function updateName(string name) returns Person {
        self.profile.name = name;
        return self.profile;
    }
    
    remote function updateCity(string city) reutns Person {
        self.profile.city = city;
        return self.profile;
    }
}
```

This will generate the following schema:

```schema
type Query {
    profile: Person!
}

type Mutation {
    updateName(name: String!): Person!
    updateCity(city: String!): Person!
}

type Person {
    name: String!
    age: Int!
    city: String!
}
```

> Note: A GraphQL schema must have a root `Query` type. Therefore, a Ballerina GraphQL service must have at least one resource function defined. 

Reviewer: ```suggestion
>**Note:** A GraphQL schema must have a root `Query` type. Therefore, a Ballerina GraphQL service must have at least one resource function defined. 
```

> We can mutate this using the following document:

Reviewer: ```suggestion
This can be mutated using the following document.
```

```
mutation updatePerson {
    updateName(name: "Mr. Lambert") {
        ... ProfileFragment
    }
    updateCity(city: "New Hampshire") {
        ... ProfileFragment
    }
}

fragment ProfileFragment on Profile {
    name
    city
}
```

> Note: This document uses two mutations, and each mutation requests same fields from the service using a fragment (`ProfileFragment`).

Reviewer: ```suggestion
>**Note:** This document uses two mutations and each mutation requests the same fields from the service using a fragment (`ProfileFragment`).
```

Result:

```json
{
    "data": {
        "updateName": {
            "name": "Mr. Lambert",
            "city": "Albuquerque"
        },
        "updateCity": {
            "name": "Mr. Lambert",
            "city": "New Hampshire"
        }
    }
}
```

> See how the result changes the `Person` record. The first mutation only changes the name, and it populates the result of the `updateName` field. Then it will execute the `updateCity` operation and populate the result. This is because the execution of mutation operations will be done serially, in the same order as they are specified in the document.

Reviewer: ```suggestion
See how the result changes the `Person` record. The first mutation changes only the name and it populates the result of the `updateName` field. Then, it will execute the `updateCity` operation and populate the result. This is because the execution of the mutation operations will be done serially in the same order as they are specified in the document.
```
